VirtoCommerce.Storefront/Domain/StaticContent/StaticContentLoader.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using VirtoCommerce.Storefront.Model.StaticContent;
+using YamlDotNet.RepresentationModel;
+
+namespace VirtoCommerce.Storefront.Domain
+{
+    public class StaticContentLoader : IStaticContentLoader
+    {
+        private static readonly Regex _headerRegExp = new Regex(@"(?s:^---(.*?)---)");
+
+        public virtual string PrepareContent(string content)
+        {
+            return RemoveYamlHeader(content);
+        }
+
+        public virtual void ReadMetaData(string content, IDictionary<string, IEnumerable<string>> metadata)
+        {
+            ReadYamlHeader(content, metadata);
+        }
+
+        private static string RemoveYamlHeader(string text)
+        {
+            var retVal = text;
+            var headerMatches = _headerRegExp.Matches(text);
+
+            if (headerMatches.Count > 0)
+            {
+                retVal = text.Replace(headerMatches[0].Groups[0].Value, "").Trim();
+            }
+
+            return retVal;
+        }
+
+        private static void ReadYamlHeader(string text, IDictionary<string, IEnumerable<string>> metadata)
+        {
+            var headerMatches = _headerRegExp.Matches(text);
+            if (headerMatches.Count == 0)
+                return;
```

Comment:
Bodies of `if`, `for`, etc. should be enclosed in braces.
Each method should have only one return at the end.